### PR TITLE
Implement element_at, ignore_elements, refactor distinct

### PR DESCRIPTION
**Pull request by godhand4826**
Where to write the test? Please help.

### CODE REVIEW

**Comment by M-Adoo** [comment]
Thanks!  Just add a test mod at the end of the file.

**Comment by godhand4826** [comment]
Thanks for you help, Great!

File: src/ops/ignore_elements.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+};
+
+#[derive(Clone)]
+pub struct IgnoreElementsOp<S> {
+  pub(crate) source: S,
+}
+
+observable_proxy_impl!(IgnoreElementsOp, S);
+
+impl<Item, Err, S, Unsub> LocalObservable<'static> for IgnoreElementsOp<S>
```
Comment (M-Adoo) [inline src/ops/ignore_elements.rs:16]: Should not only implement `LocalObservable<'static>`, use a general lifetime please.

File: src/ops/ignore_elements.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      subscription,
+    })
+  }
+}
+
+struct IgnoreElementsObserver<O> {
+  observer: O,
+}
+
+struct SharedIgnoreElementsObserver<O>(Arc<Mutex<IgnoreElementsObserver<O>>>);
```
Comment (M-Adoo) [inline src/ops/ignore_elements.rs:73]: You needn't wrap new type with `Arc` and `Rc`, just directly  implement `Observer` for `IgnoreElementsObserver` is enough.

**Review by M-Adoo — CHANGES_REQUESTED** [review]
Looks great,  thank you very much and some suggestions for you.

File: src/ops/ignore_elements.rs (hunk 2 above)
Comment (godhand4826) [reply src/ops/ignore_elements.rs:73]: Thanks for your reply!
I'm new about rust. 
For my understanding, to shared a object should wrap with Rc<_>.
To shared a object between threads should wrap with Arc<Mutex<_>>
Why here don't need to? Am I misunderstanding or missing something?

Comment (M-Adoo) [reply src/ops/ignore_elements.rs:73]: You are right. But  you can directly use `Arc` and `Rc`,   new  type `SharedIgnoreElementsObserver`  with `Arc` is needn't, same to the local version.

**Comment by godhand4826** [comment]
Re-implement ignore_elements with general lifetime. :+1:

**Review by M-Adoo — APPROVED** [review]
Looks good, thanks.

**Comment by godhand4826** [comment]
Refactor skip_last operator.
Sorry, I forgot to create branch before refactor.

**Review by M-Adoo — APPROVED** [review]
Thanks！

**Pull request merged**